tests/test-functional/test-functional-fp.h

```diff
@@ -45,7 +45,7 @@ testFunctional::testFp(const std::string& category, PimDataType dataType)
 
   // Cover scalar value testing
   // PIMeval uses uint64_t to represent bits of scalarValue
-  const float scalarValFloat = 123.0;
+  const float scalarValFloat = 123.0f;
   const uint64_t scalarVal = *reinterpret_cast<const uint64_t*>(&scalarValFloat);
```

Comment:
```suggestion
  const uint64_t scalarVal = *reinterpret_cast<const uint32_t*>(&scalarValFloat);
```

Reply:
Float type has 32 bits only. Reinterpreting it with uint64_t* can lead to some incorrect results